exp/api/v1alpha4/types.go

```diff
@@ -0,0 +1,199 @@
+	// size must be equal to or larger than the snapshot size.
+	// +optional
+	VolumeSize int64 `json:"volumeSize,omitempty"`
+
+	// The volume type
+	// For more information, see Amazon EBS Volume Types (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html)
+	// +kubebuilder:validation:Enum=standard;io1;gp2;st1;sc1;io2
+	// +optional
+	VolumeType string `json:"volumeType,omitempty"`
+}
+
+// BlockDeviceMappings specifies the block devices for the instance.
+// You can specify virtual devices and EBS volumes.
+type BlockDeviceMapping struct {
+	// The device name exposed to the EC2 instance (for example, /dev/sdh or xvdh).
+	// +kubebuilder:validation:Required
+	DeviceName string `json:"deviceName,omitempty"`
+
+	// You can specify either VirtualName or Ebs, but not both.
+	// +optional
+	Ebs EBS `json:"ebs,omitempty"`
+}
+
+// AwsLaunchTemplate defines the desired state of AWSLaunchTemplate
+type AWSLaunchTemplate struct {
+	// The ID of the launch template.
+	ID string `json:"id,omitempty"`
+
+	// The name of the launch template.
+	Name string `json:"name,omitempty"`
+
+	// The name or the Amazon Resource Name (ARN) of the instance profile associated
+	// with the IAM role for the instance. The instance profile contains the IAM
+	// role.
+	IamInstanceProfile string `json:"iamInstanceProfile,omitempty"`
+
+	// AMI is the reference to the AMI from which to create the machine instance.
+	// +optional
+	AMI infrav1.AWSResourceReference `json:"ami,omitempty"`
+
+	// ImageLookupFormat is the AMI naming format to look up the image for this
+	// machine It will be ignored if an explicit AMI is set. Supports
+	// substitutions for {{.BaseOS}} and {{.K8sVersion}} with the base OS and
+	// kubernetes version, respectively. The BaseOS will be the value in
+	// ImageLookupBaseOS or ubuntu (the default), and the kubernetes version as
+	// defined by the packages produced by kubernetes/release without v as a
+	// prefix: 1.13.0, 1.12.5-mybuild.1, or 1.17.3. For example, the default
+	// image format of capa-ami-{{.BaseOS}}-?{{.K8sVersion}}-* will end up
+	// searching for AMIs that match the pattern capa-ami-ubuntu-?1.18.0-* for a
+	// Machine that is targeting kubernetes v1.18.0 and the ubuntu base OS. See
+	// also: https://golang.org/pkg/text/template/
+	// +optional
+	ImageLookupFormat string `json:"imageLookupFormat,omitempty"`
+
+	// ImageLookupOrg is the AWS Organization ID to use for image lookup if AMI is not set.
+	ImageLookupOrg string `json:"imageLookupOrg,omitempty"`
+
+	// ImageLookupBaseOS is the name of the base operating system to use for
+	// image lookup the AMI is not set.
+	ImageLookupBaseOS string `json:"imageLookupBaseOS,omitempty"`
+
+	// InstanceType is the type of instance to create. Example: m4.xlarge
+	InstanceType string `json:"instanceType,omitempty"`
+
+	// RootVolume encapsulates the configuration options for the root volume
+	// +optional
+	RootVolume *infrav1.Volume `json:"rootVolume,omitempty"`
```

Comment:
Should this support adding non-root / secondary volumes too?

Reply:
Tracking here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2157